src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,412 @@
+       * @type {?Snowflake}
+       */
+      this.creatorId = data.creator_id;
+    } else {
+      this.creatorId ??= null;
+    }
+
+    /**
+     * The name of the guild scheduled event
+     * @type {string}
+     */
+    this.name = data.name;
+
+    if ('description' in data) {
+      /**
+       * The description of the guild scheduled event
+       * @type {?string}
+       */
+      this.description = data.description;
+    } else {
+      this.description ??= null;
+    }
+
+    /**
+     * The timestamp the guild scheduled event will start at
+     * <info>This can be potentially `null` only when it's an {@link AuditLogEntryTarget}</info>
+     * @type {?number}
+     */
+    this.scheduledStartTime = data.scheduled_start_time ? new Date(data.scheduled_start_time).getTime() : null;
+
+    /**
+     * The timestamp the guild scheduled event will end at,
+     * or `null` if the event does not have a scheduled time to end
+     * @type {?number}
+     */
+    this.scheduledEndTime = data.scheduled_end_time ? new Date(data.scheduled_end_time).getTime() : null;
+
+    /**
+     * The privacy level of the guild scheduled event
+     * @type {PrivacyLevel}
+     */
+    this.privacyLevel = GuildScheduledEventPrivacyLevels[data.privacy_level];
+
+    /**
+     * The status of the guild scheduled event
+     * @type {GuildScheduledEventStatus}
+     */
+    this.status = GuildScheduledEventStatuses[data.status];
+
+    /**
+     * The type of hosting entity associated with the scheduled event
+     * @type {GuildScheduledEventEntityType}
+     */
+    this.entityType = GuildScheduledEventEntityTypes[data.entity_type];
+
+    if ('entity_id' in data) {
+      /**
+       * The id of the hosting entity associated with the scheduled event
+       * @type {?Snowflake}
+       */
+      this.entityId = data.entity_id;
+    } else {
+      this.entityId ??= null;
+    }
+
+    if ('user_count' in data) {
+      /**
+       * The number of users who are subscribed to this guild scheduled event
+       * @type {?number}
+       */
+      this.userCount = data.user_count;
+    } else {
+      this.userCount ??= null;
+    }
+
+    if ('creator' in data) {
+      /**
+       * The user that created this guild scheduled event
+       * @type {?User}
+       */
+      this.creator = this.client.users._add(data.creator);
+    } else {
+      this.creator ??= this.client.users.resolve(this.creatorId);
+    }
+
+    if (data.entity_metadata) {
+      if ('location' in data.entity_metadata) {
+        /**
+         * The location of the event
+         * @type {?string}
+         */
+        this.location = data.entity_metadata.location;
+      } else {
+        this.location ??= null;
+      }
+    } else {
+      this.location ??= null;
+    }
+  }
+
+  /**
+   * The timestamp the guild scheduled event was created at
+   * @type {number}
+   * @readonly
+   */
+  get createdTimestamp() {
+    return SnowflakeUtil.deconstruct(this.id).timestamp;
```

Comment:
If you have rebased, you should be able to use [`SnowflakeUtil.timestampFrom`](https://discord.js.org/#/docs/main/main/class/SnowflakeUtil?scrollTo=s-timestampFrom) now:
```suggestion
    return SnowflakeUtil.timestampFrom(this.id);
```